File: examples-sdk/csharp/605-db-vilorm-crud/Main.cs
// 605-db-vilorm-crud — C# SDK equivalent
// Compile: vil compile --from csharp --input 605-db-vilorm-crud/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("vilorm-showcase", 8080);
var blog = new ServiceProcess("blog");
blog.Endpoint("POST", "/authors", "create_author");
blog.Endpoint("GET", "/authors", "list_authors");
blog.Endpoint("GET", "/posts", "list_posts");
blog.Endpoint("POST", "/posts", "create_post");
blog.Endpoint("GET", "/posts/:id", "get_post");
blog.Endpoint("PUT", "/posts/:id", "update_post");
blog.Endpoint("DELETE", "/posts/:id", "delete_post");
blog.Endpoint("POST", "/tags", "create_tag");
blog.Endpoint("GET", "/stats", "blog_stats");
server.Service(blog);
server.Compile();

File: examples-sdk/swift/605-db-vilorm-crud/main.swift
// 605-db-vilorm-crud — Swift SDK equivalent
// Compile: vil compile --from swift --input 605-db-vilorm-crud/main.swift --release

let server = VilServer(name: "vilorm-showcase", port: 8080)
let blog = ServiceProcess(name: "blog")
blog.endpoint(method: "POST", path: "/authors", handler: "create_author")
blog.endpoint(method: "GET", path: "/authors", handler: "list_authors")
blog.endpoint(method: "GET", path: "/posts", handler: "list_posts")
blog.endpoint(method: "POST", path: "/posts", handler: "create_post")
blog.endpoint(method: "GET", path: "/posts/:id", handler: "get_post")
blog.endpoint(method: "PUT", path: "/posts/:id", handler: "update_post")
blog.endpoint(method: "DELETE", path: "/posts/:id", handler: "delete_post")
blog.endpoint(method: "POST", path: "/tags", handler: "create_tag")
blog.endpoint(method: "GET", path: "/stats", handler: "blog_stats")
server.service(blog)
server.compile()
